setup.cfg

```diff
@@ -40,6 +40,7 @@ install_requires =
   astropy>=4.0
   parfive[ftp]>=1.1.0
   importlib_metadata;python_version<"3.8"
+  h5netcdf>=0.8.1
```

Comment:
Do we really want to add this as a base dependency, instead of an optional dependency, given it's only needed for new GOES files? Yes is an acceptable answer, but just want to keep dependency creep in check a bit.

Reply:
Fwiw my vote here would be no and that this is added to a `timeseries` or the like `extra_requires` category as HDF5 is a heavy dependency.

Reply:
I would also agree with this. I think @Cadair had a reason for this though? however I could be wrong.

Reply:
Going to speak again for stuart as I like to do.

The ability for time series or map to open and parse a file type it supports is core functionality.

Reply:
Nabil is basically correct with his channelling. More specifically *not* making this a core dependency is a significant change in policy. At the moment core is pretty much "the requirements needed to make map and timeseries work". I am happy to reconsider this, but we should probably do it in a dedicated issue and not on this PR.

Reply:
I opened https://github.com/sunpy/sunpy/issues/4655

Reply:
Fair enough! I feel like we've had this discussion in the past and I've agreed with this line of reasoning and then forgotten 😅